### PR TITLE
Added talking skeleton toy to annoy your co-workers with

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax, Nienhaus
rscdel: Unfortunately, the Belowyarn toys have been withdrawn from use due to Assistants choking on the batteries. Centcom apologise for the inconvenience that this lack of batteries may cause.
rscadd: We are introducing Oh-cee the skeleton figurines. Pull its string, and hear it say a variety of phrases. Now with no user removable parts!
tweak: Note that pulling the string of all talking toys is now visible to others.
/:cl:

With thanks to Nienhaus for doing the sprite.

### CODE REVIEW

**Comment by Incoming5643** [comment]
https://github.com/tgstation/-tg-station/pull/16900#issuecomment-210837823

> 1. We already have a more subtle reference for undertale (and indeed for sans and papyrus!) in the game
2. It was really blatantly a reference without any subtly, integration with the setting, or reason for existing in space station 13 outside "It's a reference!".
3. It used sounds that were indistinguishable from copyrighted sounds, a lot of quotes directly ripped from the game, and was just generally just so on the nose that it hurt.

**Comment by coiax** [comment]
1. The original sound engine was built for skeletons to use, inspired by the very reference in text spans they speak.
2. It's a toy, for children. It's cute, minimal, and for the most part, not invasive or forced.
3. Sounds are dynamically generated with a sweet engine, many of the quotes have a special theme, and I'm sorry about your nose.

But MAYBE WE SHOULD HAVE A 24 HOUR GRACE PERIOD ON REMOVAL, INSTEAD OF SPEED REVERSIONS MAYBE??????????

**Comment by Incoming5643** [comment]
It was not by my hand that my pull was merged. It was pulled here by @Cheridan who wish to pay it tribute.

**Comment by Bawhoppen** [comment]
@Cheridan close pls

**Comment by coiax** [comment]
QUICKLY, SPEED CLOSE THIS ONE TOO, BECAUSE IF YOU DON'T LIKE A PR, YOU CAN REMOVE IT WITHOUT OVERSIGHT

**Comment by Incoming5643** [comment]
I am disappointed no one took me up on a modified Symphony of the Night Dracula conversation

**Comment by Niknakflak** [comment]
Undertale is shit

**Comment by PKPenguin321** [comment]
coiax this is a horrible example of code-babying. your code is not entitled to get into the game. stop complaining like it is

**Comment by duncathan** [comment]
It is typically entitled to a 24 hour grace period, however
If cheridan closes this one don't make another PR, though

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -719,6 +719,93 @@
...
+/obj/item/toy/talking/skeleton/proc/skelly_say(message, mob/user)
+	// We want to transform any message into a list of numbers
+	// and punctuation marks
+	// For example:
+	// "Hi." -> [2, '.']
+	// "HALP GEROGE MELLONS, that swine, is GRIFFIN ME!"
+	// -> [4, 6, 7, ',', 4, 5, ',', '2', 7, 2, '!']
+	// "fuck,thissentenceissquashed" -> [4, ',', 21]
+
+	var/regex/R = regex("(\[\\l\\d]*)(\[^\\l\\d\\s])?", "g")
```
Comment (KazeEspada) [inline code/game/objects/items/toys.dm]: Reeee not 509 compatible.

Comment (RemieRichards) [inline code/game/objects/items/toys.dm]: We're not supporting 509.

Comment (KazeEspada) [inline code/game/objects/items/toys.dm]: Not until the PR is merged.

Comment (duncathan) [inline code/game/objects/items/toys.dm]: I REALLY want to speed merge it now just to shut you the fuck up

Comment (coiax) [inline code/game/objects/items/toys.dm]: Dunc, haven't we learned the lesson today that speed merging things is bad?

Comment (duncathan) [inline code/game/objects/items/toys.dm]: the temptation of pissing off kaze is difficult to overcome

**Comment by optimumtact** [comment]
Wow guys, it's two toys and you're acting like they've replaced the main players in the game with undertale.

Stop being so sensitive please.

**Comment by Incoming5643** [comment]
4. The sprites are huge, as large a spaceman when every other figure is like 10 px high
5. They add 22 individual sound files and a folder for an item with no practical use.
6. They add nearly half a meg to the download.
7. They make other toys rarer.
8. Understale has been out for over half a year already, so this isn't even timely.
9. There's absolutely no cooldown on something that generates constant noise.
10. They have absolutely no thematic relation to anything in the game.

**Comment by optimumtact** [comment]
Wow you are really really really angry over what is just two toys??????

Did undertale kill your dog or something? Spit it out.

**Comment by coiax** [comment]
The sprites were done by me, so I wouldn't call them top quality or final.
I recently added making the string pulling visible so you know who's spamming them so you can beat the shit out of them.
And I can increase the string pulling cooldown to 10 seconds if it's an issue.

**Comment by Incoming5643** [comment]
I'm not angry because it's undertale. I played undertale. It was alright, if not hilariously overrated.

I'm mildly engaged that it's so blatant and hamfisted. I love references in the game, but for the love of god you need some goddamn class and subtlety when throwing in nods like this.

Make something you have to think about to get the joke, or hide it away in some exotic situation. Don't just put it out there front and center in an unmodified form and yell "HEY LOOK IT'S THE THING FROM THE GAME!". That's crap.

Let me explain some of my old references for some comparison:

The wand of death will give you a death message from the game Nethack when you use it on yourself. Nethack is a game famous for, among other things, having a wand of death that can just ruin your day. You will almost never use a wand of death on yourself, and so this exotic situation masks an easter egg.

The D8 die is a very obscure homestuck reference. The shade of blue use for it matches exactly with a shade of blue of a set of D8s that occur in the comic (in my defense that reference was back before homestuck went to shit). No one would ever know this is a reference unless they actually read the comic, and even then they'd wonder openly if it was a reference.

The D4 die is a shade of yellow that matches up with the color scheme of an old space station 13 creature called a roro. The pips of the die also look like the closed eyes of a roro if you've ever seen the sprite. This is a meta-reference to the game as it used to be.

The admin only item the staff of nothing has the description "This staff is boring to watch because even though it came first you've seen everything it can do in other staves for years." This reference is a nod to the show Seinfeld, the "show about nothing" that created a lot of now well worn comedy tropes. For people who first see the show nowadays it might seem predictable and boring because it has been so copied. The staff of nothing is the same way, being the progenitor in the code of all wands and staves. This is a reference so obscure I don't know if anyone would have ever figured it out.

I think those are cool references, they're rare and take some thinking to grasp. Just putting a sans action figure in the game that talks in quotes from the game and makes sounds from the game is lazy, uninspired, and potentially litigious.

So yeah :-1:

**Comment by PKPenguin321** [comment]
Listen to incoming please he is very correct

**Comment by Bawhoppen** [comment]
^

**Comment by Shadowlight213** [comment]
Holy shit incoming. That's impressive.

**Comment by optimumtact** [comment]
They're so bad as references you were the only person who recognised them

**Comment by PKPenguin321** [comment]
>They're so bad as references

>bad

It's called subtlety you dork

**Comment by optimumtact** [comment]
It's not subtle when only the reference creator gets his own reference, it's nonexistent

**Comment by PKPenguin321** [comment]
That is debatable.

That's aside from the point that this PR proposes something with absolutely no subtlety, however. It's really lame and forced.

**Comment by Incoming5643** [comment]
I've had people ask me about the d8 before, I'm sure some people would get the wand of death one if they were dumb enough to self use.

But yeah the point is beating someone over the head with a really obvious reference is L A M E

**Comment by duncathan** [comment]
The d8 one and the nethack one are good IMO

**Comment by coiax** [comment]
Going to merge the skeletons into one toy, change the sprite to something more in line with the rest of the toys, and only use the dankest SS13 memes.

**Comment by Dorsisdwarf** [comment]
Meme game references :-1: 

I liked Undertale, but why? We already have a blatant and unsubtle reference in the game to the skeletons.

**Comment by coiax** [comment]
@Dorsisdwarf the PR as presented is not what I'm proposing anymore. I'm gonna close this while I work on it.

**Comment by Bawhoppen** [comment]
thought to be fair incomin, there are quite a few non-subtle references
see xenos

**Comment by Dorsisdwarf** [comment]
Xenomorphs are a cross-genre stereotype now though

**Pull request closed without merging**